controllers/api.controllers.js

```diff
@@ -53,17 +61,43 @@ exports.postNewComment = (req, res, next) => {
   const { article_id } = req.params;
   const { username, body } = req.body;
 
-  if (username === undefined) {
-    return next({ status: 400, msg: "username is required" });
+  selectArticleById(article_id)
+    .then((article) => {
+      postCommentOnArticle(article_id, username, body)
+        .then((comment) => {
+          res.status(201).send({ comment });
+        })
+        .catch((err) => {
+          next(err);
+        });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.addArticleVotes = (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+
+  if (inc_votes === undefined) {
+    return next({ status: 400, msg: "inc_votes is required" });
```

Comment:
rather than invoking next here, please do a promise rejection in your model and the use a combination of catch and next to catch and route your error to your error handling middleware in your app.